Readme and sample tweaks (#14801)
Azure/azure-sdk-for-net: sdk/keyvault/Azure.Security.KeyVault.Administration/readme.md

# Azure KeyVault Administration client library for .NET

> Azure Key Vault is a cloud service for securely storing and accessing secrets. A secret is anything that you want to tightly control 
> access to, such as API keys, passwords, or certificates. A vault is a logical group of secrets.

Reviewer: ```suggestion
Azure Key Vault is a cloud service for securely storing and accessing secrets. A secret is anything to which you want to tightly control 
access, such as API keys, passwords, or certificates. A vault is a logical group of secrets.
```

See the final two steps in the next section for details on creating the Key Vault with the Azure CLI.

### Authenticate the client

> In order to interact with the Key Vault service, you'll need to create an instance of the [KeyVaultAccessControlClient][rbac_client] class. 

Reviewer: "interact" seems a bit vague. Maybe this:
```suggestion
In order to control permissions to the Key Vault service, you'll need to create an instance of the [KeyVaultAccessControlClient][rbac_client] class. 
```

> You need a **vault url**, which you may see as "DNS Name" in the portal,  and **client secret credentials (client id, client secret, tenant id)** 

Reviewer: ```suggestion
You need a **vault URL**, which you may see as "DNS Name" in the portal,  and **client secret credentials (client id, client secret, tenant id)** 
```

```C# Snippet:HelloCreateKeyVaultBackupClient
KeyVaultBackupClient client = new KeyVaultBackupClient(new Uri(keyVaultUrl), new DefaultAzureCredential());
```

## Key concepts

### KeyVaultRoleDefinition
A `KeyVaultRoleDefinition` is a collection of permissions. A role definition defines the operations that can be performed, such as read, write, 
and delete. It can also define the operations that are excluded from allowed operations.

KeyVaultRoleDefinitions can be listed and specified as part of a `KeyVaultRoleAssignment`.

### KeyVaultRoleAssignment. 
A `KeyVaultRoleAssignment` is the association of a KeyVaultRoleDefinition to a service principal. They can be created, listed, fetched individually, and deleted.

### KeyVaultAccessControlClient
A `KeyVaultAccessControlClient` provides both synchronous and asynchronous operations allowing for management of `KeyVaultRoleDefinition` and `KeyVaultRoleAssignment` objects.

### KeyVaultBackupClient

A `KeyVaultBackupClient` provides both synchronous and asynchronous operations for performing full key backups, full key restores, and selective key restores.

> ### BackupOperation

Reviewer: Should these be H4s under `KeyVaultBackupClient` since they are related only to that? Don't really care much either way - just putting it out there.

Author: It's a good idea, but since there are so few, I decided to keep each major type at the same level.
